crates/bevy_utils/src/short_names.rs

```diff
@@ -107,4 +117,20 @@ mod name_formatting_tests {
             "do_mad_science<Test<Tube>, TypeSystemAbuse>".to_string()
         );
     }
+
+    #[test]
+    fn sub_path_after_closing_bracket() {
+        assert_eq!(
+            get_short_name("bevy_asset::assets::Assets<bevy_scene::dynamic_scene::DynamicScene>::asset_event_system"),
+            "Assets<DynamicScene>::asset_event_system".to_string()
+        );
+        assert_eq!(
+            get_short_name("(String, String)::default"),
+            "(String, String)::default".to_string()
+        );
+        assert_eq!(
+            get_short_name("[i32; 16]::default"),
+            "[i32; 16]::default".to_string()
+        );
```

Comment:
Nit: I don't know if these are technically possible. I'm pretty sure you're required to wrap them with angled brackets (i.e. `<(String, String)>::default`).